concrete/src/File/Api/FilesController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Concrete\Core\File\Api;
+
+use Concrete\Core\Api\ApiController;
+use Concrete\Core\Application\Application;
+use Concrete\Core\File\File;
+use Concrete\Core\File\FileListTransformer;
+use Concrete\Core\File\FileTransformer;
+use Concrete\Core\File\Search\SearchProvider;
+use Concrete\Core\Http\Request;
+use Concrete\Core\Permission\Checker;
+
+class FilesController extends ApiController
+{
+
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    public function __construct(Application $app, Request $request)
+    {
+        $this->app = $app;
+        $this->request = $request;
+    }
+
+    public function read($fID)
+    {
+        $fID = (int) $fID;
+        $file = File::getByID($fID);
+        if (!$file) {
+            return $this->error(t('File not found.'), 404);
+        } else {
+            $permissions = new Checker($file);
+            if (!$permissions->canViewFileInFileManager()) {
+                return $this->error(t('You do not have access to read properties about this file.'), 401);
+            }
+        }
+
+        return $this->transform($file, new FileTransformer());
+    }
+
+    public function listFiles()
```

Comment:
These methods deserve some sort of docblock

Reply:
I agree; I'll add a docblock to the read route.

Looking more at this, I'm on the fence about this. I'm not convinced this is the right way to go, and I want to make sure we follow JSON API, which I don't imagine this route does. This was part of the derek files pull request. I like the idea, but I want to think more about this and make sure we don't have to rewrite this. So I'm pulling listFiles and the FileListTransformer, and just leaving the read route.

Reply:
This would not follow JSON API as thats not what any of the current( or current as of the time when I wrote this back in december) core API things do. I was actually going to write it to follow JSON Api but it would not match any of the other things. None of our current responses follow JSON API specification.